Failure test_additional_discount_for_sales_invoice_with_discount_accounting_enabled

### Information about bug

FAIL: test_additional_discount_for_sales_invoice_with_discount_accounting_enabled (erpnext.accounts.doctype.sales_invoice.test_sales_invoice.TestSalesInvoice)

### Module

selling

### Version

Frappe - v15
Erpnext - v15

### Installation method

None

### Relevant log output / Stack trace / Full Error Message.

```shell
bench run-tests --doctype "Sales Invoice" --test test_additional_discount_for_sales_invoice_with_discount_accounting_enabled
F
======================================================================
FAIL: test_additional_discount_for_sales_invoice_with_discount_accounting_enabled (erpnext.accounts.doctype.sales_invoice.test_sales_invoice.TestSalesInvoice)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/home/aarti/postgres/frappe-bench-post/apps/erpnext/erpnext/accounts/doctype/sales_invoice/test_sales_invoice.py", line 2936, in test_additional_discount_for_sales_invoice_with_discount_accounting_enabled
    check_gl_entries(self, si.name, expected_gle, add_days(nowdate(), -1))
  File "/home/aarti/postgres/frappe-bench-post/apps/erpnext/erpnext/accounts/doctype/sales_invoice/test_sales_invoice.py", line 3902, in check_gl_entries
    doc.assertEqual(expected_gle[i][0], gle.account)
AssertionError: '_Test Account VAT - _TC' != 'Debtors - _TC'
- _Test Account VAT - _TC
+ Debtors - _TC


----------------------------------------------------------------------
Ran 1 test in 8.318s

FAILED (failures=1)
```

--- a/erpnext/accounts/doctype/sales_invoice/test_sales_invoice.py
+++ b/erpnext/accounts/doctype/sales_invoice/test_sales_invoice.py
@@ -3898,12 +3898,14 @@ def check_gl_entries(doc, voucher_no, expected_gle, posting_date, voucher_type="
 	)
 	gl_entries = q.run(as_dict=True)
 
+	expected_gle = sorted(expected_gle, key=lambda x: x[0])
+	gl_entries = sorted(gl_entries, key=lambda x: x['account'])
+
 	for i, gle in enumerate(gl_entries):
 		doc.assertEqual(expected_gle[i][0], gle.account)
 		doc.assertEqual(expected_gle[i][1], gle.debit)
 		doc.assertEqual(expected_gle[i][2], gle.credit)
 		doc.assertEqual(getdate(expected_gle[i][3]), gle.posting_date)
-
 
 def create_sales_invoice(**args):
 	si = frappe.new_doc("Sales Invoice")

